evmjit/libevmjit/Common.h

```diff
@@ -31,7 +31,7 @@ enum class ReturnCode
 
 	// Standard error codes
 	OutOfGas           = -1,
-	StackUnderflow      = -2,
+	StackUnderflow     = -2,
 	BadJumpDestination = -3,
 	BadInstruction     = -4,
 	Rejected           = -5, ///< Input data (code, gas, block info, etc.) does not meet JIT requirement and execution request has been rejected
```

Comment:
These return codes are a bit outdated. It is ongoing process but in the end JIT will not inform what kind of error happened (only OutOfGas generic abort).
So, answering your question, in case of stack size overflow and call depth you will get OutOfGas error.